Hazel/src/Platform/OpenGL/OpenGLVertexArray.cpp

```diff
@@ -50,19 +51,20 @@ namespace Hazel {
 	{
 		HZ_CORE_ASSERT(vertexBuffer->GetLayout().GetElements().size(), "Vertex Buffer has no layout!");
 
-		glBindVertexArray(m_RendererID);
-		vertexBuffer->Bind();
-
 		const auto& layout = vertexBuffer->GetLayout();
 		for (const auto& element : layout)
 		{
-			glEnableVertexAttribArray(m_VertexBufferIndex);
-			glVertexAttribPointer(m_VertexBufferIndex,
+			glEnableVertexArrayAttrib(m_RendererID, m_VertexBufferIndex);
+			glVertexArrayVertexBuffer(m_RendererID, m_VertexBufferIndex, 
+				std::dynamic_pointer_cast<OpenGLVertexBuffer>(vertexBuffer)->m_RendererID,
```

Comment:
Why the requirement to do a `dynamic_pointer_cast`? We're inside the OpenGL renderer, so we can assume we only use OpenGLVertexBuffer. With this reason, can't we do a `static_pointer_cast` instead?

```suggestion
				std::static_pointer_cast<OpenGLVertexBuffer>(vertexBuffer)->m_RendererID,
```